rust/tools/authorization-logic/transparent_release_prototype/transparent_release.auth_logic

```diff
@@ -0,0 +1,276 @@
+// that takes just the application name (e.g. "OakFunctionsLoader") as an
+// input (the names of the provenance file and endorsement file are just 
+// concatenations with the application name).
+// The output is this authorization logic code.
+
+"Verifier_OakFunctionsLoader" says {
+    "EndorsementFile_OakFunctionsLoader" canSay
+        expected_hash("ProvenanceFile_OakFunctionsLoader", some_hash) :-
+            isSymbol(some_hash).
+    
+    "EndorsementFile_OakFunctionsLoader" canSay
+        expected_hash("Binary_OakFunctionsLoader", some_hash) :-
+            isSymbol(some_hash).
+
+    "ProvenanceFileBuilder" canSay 
+        "Binary_OakFunctionsLoader" hasProvenance(some_provenance_file) :-
+            isSymbol(some_provenance_file).
+
+    "Sha256Wrapper" canSay measured_hash(some_object, some_hash) :-
+        isSymbol(some_object), isSymbol(some_hash).
+
+    "RekorLogCheck" canSay some_object canActAs "ValidRekorEntry"
+      :- isSymbol(some_object).
+
+    "Binary_OakFunctionsLoader" canActAs "OakFunctionsLoader" :-
+        "Binary_OakFunctionsLoader"
+            hasProvenance("ProvenanceFile_OakFunctionsLoader"),
+        "EndorsementFile_OakFunctionsLoader" canActAs "ValidRekorEntry",
+        expected_hash("Binary_OakFunctionsLoader", binary_hash),
+        measured_hash("Binary_OakFunctionsLoader", binary_hash),
+        expected_hash("ProvenanceFile_OakFunctionsLoader", provenance_hash),
+        measured_hash("ProvenanceFile_OakFunctionsLoader", provenance_hash).
+
+    // This we hopefully can get rid of for the real version:
+    isSymbol("6de656f11599c11548519af6b7385504ed845a4ff6600745cb3d9e174b8956af").
+    isSymbol("cb255dbbea86b128379995ff7dc805fd026e2d21c997a8df688ba9e5cb98e1c1").
+    isSymbol("ProvenanceFile_OakFunctionsLoader").
+    isSymbol("Binary_OakFunctionsLoader").
+    isSymbol("EndorsementFile_OakFunctionsLoader").
+}
+
+//-----------------------------------------------------------------------------
+// Endorsement File Wrapper
+//-----------------------------------------------------------------------------
+
+// This is an output from the wrapper for endorsement files. This 
+// wrapper is a standalone binary with a CLI that takes these arguments:
+// - a path to an endorsement file
+// - a name for the application
+// and it outputs authorization that looks like the code below
+
+// To work in comparisons in the generated authorization logic, the times need 
+// to be integers that increase as the time advances. A way to do this is 
+// to represent the time as "Unix Epoch Seconds". The wrapper for the 
+// endorsement file needs to convert the times it parses into this format. This 
+// URL gives code for doing this in many languages: https://www.epochconverter.com/
+// under the header "Convert from human-readable date to epoch".
+
+// For this output these arguments are:
+// - endorsement file --> "endorsement_file.toml"
+// - application name --> "OakFunctionsLoader"
```

Comment:
I think overall the application name being passed to all the wrappers seems redundant.

If we address each file by its hash, then we can just use the very hashes as principals. And we can have the wrappers just convert any number of files, without fear of mixing things up. For instance, there may be multiple endorsement files, or provenance files involved, and in the current design we would have to invent a new principal name for each of them.

So for instance, this could result in the following:

```
"ee3495f93e4da5e537d266a0035a981880e82374713eb1a9261114c297861bd5" says {
  ...
}
```

Where `ee3495f93e4da5e537d266a0035a981880e82374713eb1a9261114c297861bd5` is the hash of the endorsement file.

Reply:
Also, the code as written here has another way of making sure that hash checking is not skipped (without adding the `hasMeasurement/hasIdentity` features) -- The `Sha256Wrapper` is the only code trusted to give hash measurements, and the policy code looks for a claim about a measured hash from this wrapper. I think if we just write the principal names as hashes directly, we can't put this check int he policy. We are just assuming that the hashes dropped in by each one of the wrappers is just right (i.e. we have to trust every one of these wrappers to write its hash correctly instead of just trusting a wrapper that just calls sha).

Reply:
@tiziano88 any followup thoughts?

For moving towards the MVP:
Since I think we would need an extra feature to do this, and I think this doesn't effect the rest of the engineering work, are you fine with just using strings as principals and another wrapper to check the hashes until the rest of the work to reach this milestone is completed? I think we can easily revisit this after we reach that point.

Longer term:
What do you think about the point about the ShaWrapper or the hasMeasurement/hasIdentity features preventing hash checking from being skipped?  (I guess the hasMeasurement/hasIdentity feature confused everyone, and also introduces a domain specific feature in the core language, so maybe it's not good).

I guess if the top level computes the hashes of the objects and fills in these names as the hashes, the piece of code trusted to do hash checking is still the relatively small top level. I think we can't do this for every principal or piece of policy code though because there is no way to get a "known good hash" for the `Verifier_OakFunctionsLoader` (the main policy code), so we would still need to allow for some principal names that are not hashes.

@rbehjati @bgogul any thoughts?

Reply:
Opened https://github.com/google-research/raksha/issues/367

Reply:
> It will make policies much harder to read and more brittle in the sense that there would be no way to specify e.g., that a set of hashes are all fine.

I very much agree with this point :) 

But also, the application name is already included in the endorsement file. So probably it does not need to be a separate input to the wrapper. 

I also very much agree with `hasMeasurement/hasIdentity` being confusing. So, I vote for avoiding that :)

Reply:
(FYI the idea of using hashes as principal names for some of the principals did grow on me a bit after reading this initially, but I think only if we always immediately map them into other names (like `"EndorsementFile_AppName"` but doing that mapping would require that extra feature.)